src/StatsdClient/Transport/NamedPipeTransport.cs

```diff
@@ -0,0 +1,101 @@
+#if NAMED_PIPE_AVAILABLE
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Threading;
+
+namespace StatsdClient.Transport
+{
+    internal class NamedPipeTransport : ITransport
+    {
+        private readonly NamedPipeClientStream _namedPipe;
+        private readonly TimeSpan _timeout;
+        private byte[] _internalbuffer = new byte[0];
+
+        // As SpinLock is a struct, if it is marked as `readonly`, each time it is used
```

Comment:
Can we reword this slightly:
```
 As SpinLock is a struct, each time it is used a new copy is created, if the field is marked `readonly` this leads to the error: "System.Threading.SynchronizationLockException : The calling thread does not hold the lock", because the variable is being mutated. 
```
Or something like that, this is definitely a useful comment, but I was slightly confused by the current wording.